### PR TITLE
XML Schema for EBML Schema

### PR DESCRIPTION
**Pull request by dericed**
allows validation of EBML Schemas

### CODE REVIEW

File: EBMLSchema.xsd [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+<xsd:schema xmlns:xsd="http://www.w3.org/2001/XMLSchema" elementFormDefault="qualified" version="01">
+  <xsd:element name="EBMLSchema" type="EBMLSchemaType"/>
+  <xsd:complexType name="EBMLSchemaType">
+    <xsd:annotation>
+      <xsd:documentation/>
+    </xsd:annotation>
+    <xsd:sequence>
+      <xsd:element name="element" type="elementType" minOccurs="0" maxOccurs="unbounded"/>
+    </xsd:sequence>
+    <xsd:attribute name="docType" use="required" />
```
Comment (retokromer) [inline EBMLSchema.xsd]: Cosmetic: <xsd:attribute name="docType" use="required"/>
(no space before closing empty tag)

File: EBMLSchema.xsd [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    <xsd:attribute name="version" use="required" />
```
Comment (retokromer) [inline EBMLSchema.xsd]: ditto

**Comment by dericed** [comment]
I forced an update to respond to @retokromer's comments. I also removed the <xsd:documentation> elements for now. These would be to offer definitions of the elements of the EBML Schema, but I'd rather wait or just reference the EBML specification to learn about what the elements means.

Other issues with this:
- Matroska's ebml_schema.xml uses local attributes (`cppname` and `webm`) we need to allow them or remove them.
- Should EBMLSchema.xsd have a namespace? If so, what?

File: EBMLSchema.xsd [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  </xsd:complexType>
+  <xsd:complexType name="elementType">
+    <xsd:sequence>
+      <xsd:element name="documentation" type="documentationType" minOccurs="0" maxOccurs="unbounded"/>
+    </xsd:sequence>
+    <xsd:attribute name="name" use="required"/>
+    <xsd:attribute name="path" use="required"/>
+    <xsd:attribute name="id" use="required"/>
+    <xsd:attribute name="minOccurs" default="0"/>
+    <xsd:attribute name="maxOccurs" default="1"/>
```
Comment (robUx4) [inline EBMLSchema.xsd]: With this [PR](https://github.com/Matroska-Org/ebml-specification/pull/124) the default for `maxOccurs` is like `EBMLMaxOccurrence` it's infinite.

**Comment by retokromer** [comment]
@dericed In the old days to have a namespace was considered good practice. I didn't follow any discussion about pros and cons of this. Intuitively I would answer yes, and I would chose «something» including ietf and cellar, like `http://ietf.org/cellar/ebml`.

**Comment by dericed** [comment]
Added a namespace based on @retokromer's suggestion, but with an `https` version.

**Comment by dericed** [comment]
If this PR is accepted, I'll update the ebml_matroska.xml file in the matroska spec repo.

**Comment by retokromer** [comment]
LGTM

**Comment by mbunkus** [comment]
LGTM, too. Maybe you could squash both commits and remove the `[WIP]` from the commit message; it might be confusing to let it in.

**Comment by dericed** [comment]
Updated to @mbunkus's recommendations. Please re-review.

**Comment by retokromer** [comment]
LGTM

**Pull request merged**